internal/dispatcher/nats.go

```diff
@@ -0,0 +1,40 @@
+package dispatcher
+
+import (
+	"log"
+
+	"github.com/nats-io/nats.go"
+)
+
+type NatsDispatcher struct {
+	connection *nats.Conn
+	connError  error
+}
+
+func (nd *NatsDispatcher) Publish(topic string, data []byte) error {
+	nc := nd.connection
+	nc.Publish(topic, data)
+	nc.Flush()
+	if err := nc.LastError(); err != nil {
+		log.Fatal(err)
+	} else {
+		log.Printf("Published [%s] : '%s'\n", topic, data)
+	}
+	return nil
+}
+
+func (nd *NatsDispatcher) Subscribe(topic string, handler nats.MsgHandler) error {
```

Comment:
be careful here on the interface, we may run into conflicts, i'll check it

Reply:
yes, it seems nats.io uses it own struct for `MsgHandler`